### PR TITLE
Translating ch15-02 in French.

### LINKED ISSUE #160: Translating ch15-02
I'm working on the French translation of this page.

### PR DESCRIPTION
**Pull request by Jimskapt**
Closes #160

### CODE REVIEW

File: FRENCH/src/ch15-02-deref.md [hunk 1]
```diff
@@ -0,0 +1,741 @@
...
+or smart pointers.
+-->
+
+Regardons d'abord comment l'opérateur de déréférencement fonctionne avec des
+références classiques. Ensuite nous essayerons de définir un type personnalisé
+qui se comporte comme `Box<T>`, et voir pourquoi l'opérateur de déréférencement
+ne fonctionne pas comme une référence sur notre type fraîchement défini. Nous
+allons découvrir comment implémenter le trait `Deref` de manière à ce qu'il soit
+possible que les pointeurs intelligents fonctionnent comme les références.
+Ensuite nous verons la fonctionnalité d'*extrapolation de déréférencement* de
```
Comment (yozhgoor) [inline FRENCH/src/ch15-02-deref.md:38]:
```suggestion
Ensuite nous verrons la fonctionnalité d'*extrapolation de déréférencement* de
```

File: FRENCH/src/ch15-02-deref.md [hunk 2]
```diff
@@ -0,0 +1,741 @@
...
+Rust et voir comment elle nous permet de travailler à la fois avec des
```
Comment (yozhgoor) [inline FRENCH/src/ch15-02-deref.md:39]:
```suggestion
Rust et comment elle nous permet de travailler à la fois avec des
```

File: FRENCH/src/ch15-02-deref.md [hunk 3]
```diff
@@ -0,0 +1,741 @@
...
+Le troisième cas est plus ardue : Rust va aussi procéder à une extrapolation de
+déréférencement d'une référence immuable vers une référence mutable. A cause des
+règles d'emprunt, si vous avez une référence mutable, cette référence mutable
+doit être la seule référence vers cette donnée (autrement, le programme ne peut
+pas être compilé). Convertir une référence mutable vers une référence immuable
+ne va jamais casser les règles d'emprunt. Convertir une référence immuable vers
+une référence mutable nécessite que la référence immuable initiale soit la seule
+référence immuable vers cette donnée, mais les règles d'emprunt n'empêchent pas
+cela. Ainsi, Rust ne peut pas déduire que la conversion d'une référence immuable
+vers une référence mutable soit possible.
```
Comment (yozhgoor) [inline FRENCH/src/ch15-02-deref.md:741]:
```suggestion
Le troisième cas est plus ardu : Rust va aussi procéder à une extrapolation de
déréférencement d'une référence mutable vers une référence immuable. Mais l'inverse
est *impossible*: une extrapolation de déréférencement d'une valeur immuable ne 
donnera jamais une référence mutable. A cause des règles d'emprunt, si vous avez une 
référence mutable, cette référence mutable doit être la seule référence vers cette donnée 
(autrement, le programme ne peut pas être compilé). Convertir une référence mutable vers 
une référence immuable ne va jamais casser les règles d'emprunt. Convertir une référence 
immuable versune référence mutable nécessite que la référence immuable initiale soit la seule
référence immuable vers cette donnée, mais les règles d'emprunt n'empêchent pas cela. 
Ainsi, Rust ne peut pas déduire que la conversion d'une référence immuablevers une 
référence mutable soit possible.
```

Comment (Jimskapt) [reply FRENCH/src/ch15-02-deref.md]: 0ula oui bien vu ! :-)

**Pull request merged**